lib/internal/util/inspect.js

```diff
@@ -2101,10 +2098,7 @@ function format(...args) {
 }
 
 function formatWithOptions(inspectOptions, ...args) {
-  if (typeof inspectOptions !== 'object' || inspectOptions === null) {
-    throw new ERR_INVALID_ARG_TYPE(
-      'inspectOptions', 'object', inspectOptions);
-  }
+  validateObject(inspectOptions, 'inspectOptions');
```

Comment:
```suggestion
  validateObject(inspectOptions, 'inspectOptions', { allowArray: true });
```
Is there any reason not to sidestep the breakage?

Reply:
Previously `util.inspectWithOptions([], "str")` prints `''` and it throws `ERR_INVALID_ARG_TYPE`, which could be a breakage in some package as you said.
Ideally this should not allow `Array` but at this time, I don't have any reason not to sidestep the breakage. Thanks for your suggestion.